python/ray/_raylet.pyx

```diff
@@ -480,42 +480,50 @@ cdef execute_task(
         const c_vector[CObjectID] &c_return_ids,
         c_vector[shared_ptr[CRayObject]] *returns):
 
-    worker = ray.worker.global_worker
-
-    actor_id = worker.core_worker.get_actor_id()
-    job_id = worker.core_worker.get_current_job_id()
-    task_id = worker.core_worker.get_current_task_id()
+    cdef object worker = ray.worker.global_worker
+    cdef object manager = worker.function_actor_manager
+    cdef dict execution_infos = manager.execution_infos
+    cdef CoreWorker core_worker = worker.core_worker
+    cdef JobID job_id = core_worker.get_current_job_id()
+    cdef CTaskID task_id = core_worker.core_worker.get().GetCurrentTaskId()
 
     # Automatically restrict the GPUs available to this task.
     ray.utils.set_cuda_visible_devices(ray.get_gpu_ids())
 
-    function_descriptor = FunctionDescriptor.from_bytes_list(
-        ray_function.GetFunctionDescriptor())
+    descriptor = tuple(ray_function.GetFunctionDescriptor())
 
     if <int>task_type == <int>TASK_TYPE_ACTOR_CREATION_TASK:
-        actor_class = worker.function_actor_manager.load_actor_class(
-            job_id, function_descriptor)
+        function_descriptor = FunctionDescriptor.from_bytes_list(
+            ray_function.GetFunctionDescriptor())
+        actor_class = manager.load_actor_class(job_id, function_descriptor)
+        actor_id = core_worker.get_actor_id()
         worker.actors[actor_id] = actor_class.__new__(actor_class)
         worker.actor_checkpoint_info[actor_id] = (
             ray.worker.ActorCheckpointInfo(
                 num_tasks_since_last_checkpoint=0,
                 last_checkpoint_timestamp=int(1000 * time.time()),
                 checkpoint_ids=[]))
 
-    execution_info = worker.function_actor_manager.get_execution_info(
-        job_id, function_descriptor)
+    execution_info = execution_infos.get(descriptor)
+    if not execution_info:
+        function_descriptor = FunctionDescriptor.from_bytes_list(
+            ray_function.GetFunctionDescriptor())
+        execution_info = manager.get_execution_info(job_id, function_descriptor)
+        execution_infos[descriptor] = execution_info
+
     function_name = execution_info.function_name
-    extra_data = {"name": function_name, "task_id": task_id.hex()}
+    extra_data = (b'{"name": ' + function_name.encode("ascii") +
+                  b' "task_id": ' + task_id.Hex() + b'}')
 
     if <int>task_type == <int>TASK_TYPE_NORMAL_TASK:
-        title = "ray_worker:{}()".format(function_name)
+        title = "ray_worker:" + function_name + "()"
```

Comment:
Is this substantially faster than `.format()`? If this doesn't make a big difference I don't think we should change it

Reply:
Or maybe there's a cython format equivalent?

Reply:
done, shouldn't be a big deal